lib/openhab/dsl/items/items.rb

```diff
@@ -26,19 +30,23 @@ class Items < SimpleDelegator
         # @param name [String] Item name to check
         # @return [Boolean] true if the item exists, false otherwise
         def include?(name)
-          # rubocop: disable Style/GlobalVars
-          !$ir.getItems(name).empty?
-          # rubocop: enable Style/GlobalVars
+          !$ir.getItems(name).empty? # rubocop: disable Style/GlobalVars
         end
         alias key? include?
+
+        def each(&block)
+          # ideally we would do this lazily, but until ruby 2.7
```

Comment:
Even in 2.7 will need to look very closely.  I think ruby uses fibers for lazy, which converted into threads for JRuby.

Reply:
doesn't ruby implement _any_ block-to-enumerator conversion (i.e. `get_enum` or `to_enum`) via fibers?

and then we get into my lack of JVM familiarity - is it necessarily bad to spin up threads? OpenHAB already has a gazillion of them!

Reply:
I did a bit of reading. tl;dr: JRuby still has problems with fibers, particularly with lazy enumerators. (see https://github.com/jruby/jruby/issues/6773). there have been multiple references to using Loom to re-implement fibers as true JVM native coroutines, but afaict that's still aspirational. given jruby _just_ released ruby 2.6 support, it could be a while before 2.7, and then we can re-evaluate